ijroi/ijroi.py

```diff
@@ -89,17 +87,33 @@ def getfloat():
     stroke_color = get32()
     fill_color = get32()
     subtype = get16()
-    if subtype != 0:
-        raise NotImplementedError('roireader: ROI subtype %s not supported (!= 0)' % subtype)
+    
     options = get16()
     arrow_style = get8()
     arrow_head_size = get8()
     rect_arc_size = get16()
     position = get32()
     header2offset = get32()
 
+    # End Header data
+    #===========================================================================
+
+    #RoiDecoder.java#L177
+    subPixelResolution = ((options&SUB_PIXEL_RESOLUTION)!=0) and (version>=222)
+    
+    # Check exceptions
+
+    if magic != b'Iout':
+        raise ValueError('Magic number not found')
```

Comment:
If the magic doesn't match we should fail immediately instead of reading any more, since we don't expect anything else to make sense if this isn't true

Reply:
Returned the magic check back to the top, sorry was a little over eager with the reformat. 69382d3d8dddade15263987db6af47e9323b7443